host_applications/linux/apps/gencmd/gencmd.c

```diff
@@ -50,27 +64,38 @@ int main( int argc, char **argv )
            argv++;
            argc--;
        }
+   } else {
+      // no arguments passed, so show basic usage
+      show_usage();
    }
 
    vcos_init();
 
     if ( vchi_initialise( &vchi_instance ) != 0)
     {
-        printf( "VCHI initialization failed\n" );
+        fprintf( stderr, "VCHI initialization failed\n" );
         return -1;
     }
 
     //create a vchi connection
     if ( vchi_connect( NULL, 0, vchi_instance ) != 0)
     {
-        printf( "VCHI connection failed\n" );
+        fprintf( stderr, "VCHI connection failed\n" );
         return -1;
     }
 
     vc_vchi_gencmd_init(vchi_instance, &vchi_connection, 1 );
 
     if (argc > 1)
     {
+       // first check if we were invoked with either -h or --help
+       // in which case show basic usage and exit
+      if( ( strcmp( argv[1], "-h" ) == 0) || strcmp( argv[1], "--help" ) == 0 )
```

Comment:
Minor inconsistency in that the condition to the left of the `||` has an extra pair of brackets than the condition to the right of the `||`.